core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -587,6 +615,23 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         }
     }
 
+    internal fun setReligionBasedModifier() {
+        val shareReligionTrueFalseOrNull = civInfo.getDiplomacyManager(otherCiv()).believesSameReligion()
+        if (shareReligionTrueFalseOrNull!=null) {
```

Comment:
Since null and false are treated the same this will turn into a simple if/else

Reply:
OK thank you!
About the null, you are totally right! I had put a distinction between true, false and null to be eventually able to use the same function to provide negative modifiers if the majority religions are different (not if there is no majority religion at all), that's why I had implemented it like this. 
Then I realized that was not the case of a negative modifier as well but just left it like this.
I don't know if it can be useful in the future for other more detailed diplomatic feature, but for now I agree, this granularity is not needed.
Thank you, I will fix it!